index.html

```diff
@@ -3,134 +3,481 @@
+    if (options.region !== 'all') {
+      var region = options.region;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.Region === region;
       });
     }
 
+ // HISTOGRAM Y
+     var yhist = d3.layout.histogram()
+      .bins(chart.y.ticks())
+      .value(function (d) {return d[chart.yvariable];})
+      (txData);  
+    
+
+      //Y Histogram's "Height" scale which is actually horizontal
+     var hx = d3.scale.linear()
+      .domain([0, d3.max(yhist, function (d) { return d.length; })])
+      .range([0, height/barMax]);
+
+
+    var ybar = chart.svg.selectAll('.ybar')
+      .data(yhist)
+
+
+    //this is coming out way too big. dx for yhist is 2000, which is the correct value. The way I understand this syntax is that its converting that 2000 to a pixel value to fit along my earlier defined y scale.  This is the way it seemed to work for the x axis histogram, and also in calculating the position of each bar 
+
+    //Hacked this by dividing by 12. I'm still unsure why without the /12 it didnt come out to the right height.
+    var ybarHeight = chart.y(yhist[1].x-yhist[0].x)/(chart.y.ticks().length-1)+1.8;
+
+    var ybarSpace = ybarHeight/4;
+    //Positioning seems to be almost there, except its offset by one bar's length
+    //Update: Hacked this too by subtracting one bar height. 
+    
+    //ENTER
+    ybar
+      .enter().append('rect')
+      .attr('class', 'ybar')
+      .attr('height', ybarHeight-(1/2)*ybarHeight)
+      .attr('y', function (d) {return (chart.y(d.x)-ybarHeight) + ybarSpace;})
+      .on('mouseover', chart.barTooltip.show)
+      .on('mouseout', chart.barTooltip.hide);
+     
+
+    ybar
+      .transition().duration(1000).delay(150)
+      .attr('width', function (d) {return hx(d.length);})  //y scale
+      .attr('x', 0);
+
+
+   // HISTOGRAM X
+    var hist = d3.layout.histogram()
+      .bins(chart.x.ticks())
+      .value(function (d) {return d[chart.xvariable];})
+      (txData);
+
+    var hy = d3.scale.linear()
+      .domain([0, d3.max(hist, function (d) { return d.length; })])
+      .range([0, height/barMax]);
+
+    var bar = chart.svg.selectAll('.bar')
+      .data(hist)
+
+    var barWidth = chart.x(hist[0].dx)+1;
+    var xbarSpace = barWidth/4;
+   
+
+    bar
+      .enter().append('rect')
+      .attr('class', 'bar')
+      .attr('width', barWidth-(1/2)*barWidth)
+      .attr('x', function (d) {return (chart.x(d.x))+ xbarSpace;})
+      .on('mouseover', chart.barTooltip.show)
+      .on('mouseout', chart.barTooltip.hide);
+
+    bar
+      .transition().duration(1000).delay(150)
+      .attr('height', function (d) {return hy(d.length);})  //y scale
+      .attr('y', function (d) {return height - hy(d.length);});
+
+       
+    //No second chart histogram yet. Will work on converting to the constructor pattern.
+     
+
     // SCATTERPLOT
+    var points = chart.svg.selectAll('.point')
+      .data(txData);
+
```

Comment:
So you changed one of the two things to make the transitions work really well - I did notice the fade in fade out for the first time they appear.
 
Note that by default, every time the update function runs, it's transitioning the circles using the index. So the circle corresponding to the first row of data in the prior subset is becoming the circle corresponding to the first row of the new subset.

This doesn't really make that much sense for this data, since these transitions generally indicate connectedness (one thing moving between various conditions). You have circles that represent one state becoming circles representing another state.

You might instead want:
```js
.data(txData, function(d) { return d.unique_key } )
```
Where unique_key was a column of data that's unique for each row of data. This way d3 will know not to transition the circles form one place to another (despite that looking cool) rather than just having them fade in and out.